vars/main.yml

```diff
@@ -0,0 +1,11 @@
+# SPDX-License-Identifier: MIT
+---
+# Put the role's internal variables here that are not distribution specific.
+
+__certificate_provider_default: certmonger
+
+__certificate_provider_vars:
```

Comment:
I would suggest to
- not nest variables inside dictionaries - a flatter structure is usually better
- load provider-specific variables from separate files, just as you do for distribution-version.

Reply:
My bad, I did not realize there can be several providers at once - given that, your approach is probably the best one possible.